fendermint/vm/message/src/signed.rs

```diff
@@ -104,14 +110,22 @@ impl SignedMessage {
         // so at least for now they are easy to tell apart: any `f410` address is coming from Ethereum API and must have
         // been signed according to the Ethereum scheme, and it could not have been signed by an `f1` address, it doesn't
         // work with regular accounts.
+        //
+        // We detect the case where the recipient is not an ethereum address. If that is the case then use regular signing rules,
+        // which should allow messages from ethereum accounts to go to any other type of account, e.g. custom Wasm actors.
         match maybe_eth_address(&message.from) {
-            Some(addr) => {
+            Some(addr) if is_eth_addr_compat(&message.to) => {
```

Comment:
What will `message.to` be here in case of a contract deployment? Does some other component earlier in the system inject an address? (it's empty in the original Ethereum transaction)

Reply:
The EAM address is injected in [to_fvm_message](https://github.com/consensus-shipyard/ipc/blob/ae750fe9778d2b6dda10768c3b6966ba3c648238/fendermint/vm/message/src/conv/from_eth.rs#L26-L27) which is called by [send_raw_transaction](https://github.com/consensus-shipyard/ipc/blob/ae750fe9778d2b6dda10768c3b6966ba3c648238/fendermint/eth/api/src/apis/eth.rs#L621) in the API facade. The opposite is what is happening in [to_eth_address](https://github.com/consensus-shipyard/ipc/blob/ae750fe9778d2b6dda10768c3b6966ba3c648238/fendermint/vm/message/src/conv/from_fvm.rs#L43-L44).

Reply:
Makes sense. Thanks!

Reply:
So this address will be an ID address, which is Eth address compatible; that's how we match this arm.